cl/search/tests/tests_es_parenthetical.py

```diff
@@ -0,0 +1,737 @@
+import datetime
+import operator
+from datetime import date
+from functools import reduce
+
+from django.urls import reverse
+from elasticsearch_dsl import Q
+from lxml import html
+
+from cl.lib.elasticsearch_utils import (
+    build_daterange_query,
+    build_es_filters,
+    build_es_main_query,
+    build_fulltext_query,
+    build_sort_results,
+    build_terms_query,
+    group_search_results,
+)
+from cl.people_db.factories import PersonFactory
+from cl.search.documents import (
+    ParentheticalGroupDocument,
+    parenthetical_group_index,
+)
+from cl.search.factories import (
+    CitationWithParentsFactory,
+    CourtFactory,
+    DocketFactory,
+    OpinionClusterFactory,
+    OpinionsCitedWithParentsFactory,
+    OpinionWithParentsFactory,
+    ParentheticalFactory,
+    ParentheticalGroupFactory,
+)
+from cl.search.models import PRECEDENTIAL_STATUS, SEARCH_TYPES, Citation
+from cl.tests.cases import ESIndexTestCase, TestCase
+
+
+class ParentheticalESTest(ESIndexTestCase, TestCase):
+    """Parenthetical ES search related tests"""
+
+    @classmethod
+    def setUpTestData(cls):
+        # Create a unique index for this test case
+        parenthetical_group_index._name = cls.__name__.lower()
+        cls.rebuild_index("search.ParentheticalGroup")
```

Comment:
This is neat, but shouldn't it go on the ESIndexTestCase instead?

Reply:
Yeah, I thought about it, however the index to use in test classes will depend on the test on the class.

For example:

If running test related to OA the index name will be: `oral_arguments_index`

So I think it's not possible to automatically set the index name in `ESIndexTestCase` since this class is expected to be used on different documents. So it should be manually set in Test classes as required.

Reply:
Wouldn't it be fine in that case if it was dynamically named?

Reply:
Correct, I changed the method to generate a unique name for each index on tests. I created a new file `es_indices` in order to register all our indices. 

Also we should add each index to:
`es_indices_registered = [parenthetical_group_index]`

So that in `ESIndexTestCase` we set an unique name (based on the test class) for each ES index registered in `es_indices_registered` ensuring that all the indices get a unique name for testing purposes.